internal/bundle/file.go

```diff
@@ -0,0 +1,57 @@
+package bundle
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+const (
+	markerFileName = ".bundle"
+)
+
+var execDir string
+
+func executableDirectory() string {
+	if execDir != "" {
+		return execDir
+	}
+
+	// Get the absolute path of the executable
+	executablePath, err := os.Executable()
+	if err != nil {
+		panic(fmt.Sprintf("Failed determining executable path: %v\n", err))
+	}
+
+	// Resolve the absolute path (eliminate any symbolic links)
+	absolutePath, err := filepath.EvalSymlinks(executablePath)
+	if err != nil {
+		panic(fmt.Sprintf("Failed determining executable path: %v\n", err))
+	}
+
+	execDir = filepath.Dir(absolutePath)
+	return execDir
+}
+
+func IsBundled() bool {
+	markerPath := GetPath(markerFileName)
+	_, err := os.Stat(markerPath)
+	return err == nil
+}
+
+func GetPath(path string) string {
+	return filepath.Join(executableDirectory(), path)
+}
+
+func ReadFileBytes(path string) ([]byte, error) {
+	return os.ReadFile(GetPath(path))
+}
+
+func ReadFile(path string) (io.Reader, error) {
+	return os.Open(GetPath(path))
```

Comment:
files are not closed

Reply:
Good point, options:

1. Leave it as is because when files are garbage collected - they are closed, see [this](https://go-review.googlesource.com/c/go/+/256899) and [this](https://github.com/golang/go/issues/41505). Also note that we are not inside some long running application which will require restarts and FDs are going to be closed right after process termination anyway

3. Expose *os.File object instead of plain io.Reader, personally don't see any benefits out of it in the real world

Reply:
`SetFinalizer` is not so reliable thing, but yeah it's gonna be closed when cli is existed. Still, some `go vet` can complain :) nvm